plugins/api/src/set_grid_configuration_value.cpp

```diff
@@ -37,10 +43,40 @@ namespace
 
     auto rs_set_grid_configuration_value(RsComm*, const GridConfigurationInput*) -> int;
 
+    auto is_grid_configuration_allowed_to_be_modified(const char* _namespace, const char* _option_name) -> bool;
+
     //
     // Function Implementations
     //
 
+    auto is_grid_configuration_allowed_to_be_modified(const char* _namespace, const char* _option_name) -> bool
+    {
+        constexpr auto protected_config_list = std::to_array<std::pair<std::string_view, std::string_view>>(
+            {{"database", "schema_version"}, {"delay_server", ""}});
+
+        // If none of the namespace/option name combinations in the protected list match the passed in namespace and
+        // option name, it is allowed to be modified and so the function should return true. If any instance returns
+        // true, that means a match was found in the list of protected configurations, so we should return false in this
+        // function because that means the grid configuration is not allowed to be modified.
+        return std::none_of(std::begin(protected_config_list),
+                            std::end(protected_config_list),
+                            [&_namespace, &_option_name](const auto& _pair) {
+                                const auto& protected_namespace = _pair.first;
+                                const auto& protected_option_name = _pair.second;
```

Comment:
Structured bindings?

Not required to change anything. Just highlighting it.

I always wonder if it's better to copy the string views rather than take a reference. We should measure the difference one day.

Reply:
I feel like... I meant to use structured bindings, but never revisited this.

Reply:
Not really. It's not urgent. Of course, you're free to look into it if you want.

I've just wondered which is better in general. The design of string_view implies copying is best, but there may be cases where passing by reference may be better.

Hmm, saying that out loud makes me think copying is always better for string_view since passing it by reference introduces one more level of indirection.

Reply:
Well, would you look at that...

https://quick-bench.com/q/LhApqVtkVv5BthpOiF19FeMyXEI

Reply:
If i'm reading that link correctly... copy is slightly faster.... but then changing to `-O3` makes them equivalent.

Reply:
Looks like they are nearly identical in performance.

Reply:
I think this is okay as-is given the bench test results. Resolving.